alerter.c

```diff
@@ -11,16 +15,26 @@ int networkAlertStub(float celcius) {
     return 200;
 }
 
+#if(PRODUCTION_ENABLED != PRODUCTION_ON)
+void testalertInCelcius(float celcius, int returnCode) {
+    assert(celcius<returnCode);
+}  
+#endif
+
 void alertInCelcius(float farenheit) {
     float celcius = (farenheit - 32) * 5 / 9;
     int returnCode = networkAlertStub(celcius);

```

Comment:
Production code should not be aware of test code. Test code should be abstracted neatly via an interface or a function pointer.
There is a danger of accidental execution of test code in field!!